recipe/2024-04-terraform-provider-secrets.md

```diff
@@ -0,0 +1,263 @@
+This document  describes in detail the handling of secrets data between the Engine and Driver for Terraform Providers building on the above designs.
+
+## Terms and definitions
+
+| Term     | Definition                                                                                                                                                                                                 |
+| -------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Terraform Provider | A Terraform provider is a plugin that allows Terraform to interact with and manage resources of a specific infrastructure platform or service, such as AWS, Azure, or Google Cloud. |
+
+## Objectives
+
+**Reference for new type recipeConfig and handling of secrets for Private Terraform Repository:** [Design document for Private Terraform Repository](https://github.com/radius-project/design-notes/blob/3644b754152edc97e641f537b20cf3d87a386c43/recipe/2024-01-support-private-terraform-repository.md)
+
+**Secret Implementation for Private Terraform Repository:**[Secret Implementation for Private Terraform Repository](https://github.com/radius-project/radius/pull/7306)
+
+**Reference for support of multiple Terraform Providers in Radius:** [Design Document to support multiple Terraform Providers](https://github.com/radius-project/design-notes/pull/39/files)
+
+> **Issue Reference:** <!-- (If appropriate) Reference an existing issue that describes the feature or bug. -->
+https://github.com/radius-project/radius/issues/6539
+
+### Goals
+
+Describe data flow, specific function calls when using secrets to configure Terraform Providers, building on design in the referred design documents.
+
+### Non goals
+
+The document is focussed on handling secrets for Terraform Providers, with `Applications.Core/secretStores` being the underlying resource for the secret store (which today supports Kubernetes secrets). Other sources and types of `secretsStores` are out of scope for this document. 
+
+### User scenarios (optional)
+
+#### User story 1
+As an operator, I maintain a set of Terraform recipes for use with Radius. I have a set of provider configurations which are applicable across multiple recipes. These provider configurations now include secrets, which are injected into the Terraform configuration, allowing the Terraform process to access them as needed.
+
+
+## User Experience (if applicable)
+
+**Sample Input:**
+
+This is pulled in from existing document:
+[Design Document to support multiple Terraform Providers](https://github.com/radius-project/design-notes/pull/39/files)
```

Comment:
Is review this doc required for someone to understand the changes in this section? If not, then I'd remove this reference and call out the additions being proposed in this doc to make it easy to understand.